### PR TITLE
Merge michael's branch

### PR DESCRIPTION
**Pull request by michaelliangau**

### CODE REVIEW

File: .gitignore
Comment (blakedehaas) [inline .gitignore:1]: ignoring data, visualizations, and autogenerated files

File: dataset/visualizations/dataset_visualization.py
Comment (blakedehaas) [inline dataset/visualizations/dataset_visualization.py:1]: Tool to look at dataset before processing

File: evaluate.py [hunk 1]
```diff
@@ -13,61 +13,118 @@
...
+output_size = 150
 model = FF_2Network(input_size, hidden_size, output_size).to("cuda")
 model.load_state_dict(torch.load(f'checkpoints/{model_name}.pth'))
 model.eval()  # Set the model to evaluation mode
 
 # Load dataset
-test_ds = Dataset.load_from_disk("data/akebono_solar_combined_v6_chu_val")
-test_ds = test_ds.remove_columns(['DateTimeFormatted', 'Ne1', 'Pv1', 'Te2', 'Ne2', 'Pv2', 'Te3', 'Ne3', 'Pv3', 'I1', 'I2', 'I3'])
+test_ds = Dataset.load_from_disk("data/akebono_solar_combined_v7_chu_test")
+test_ds = test_ds.remove_columns(['Te1', 'Pv1', 'Te2', 'Ne2', 'Pv2', 'Te3', 'Ne3', 'Pv3', 'I1', 'I2', 'I3'])
```
Comment (blakedehaas) [inline evaluate.py:34]: Testing on Ne1 to see if our dataset works with a simpler problem (change Te1 to Ne1 when testing electron temperature)

File: evaluate.py [hunk 2]
```diff
@@ -13,61 +13,118 @@
...
-output_columns = ['Te1']
+output_columns = ['Ne1']
 all_columns = input_columns + output_columns
 
 # Model
 input_size = len(input_columns)
 hidden_size = 2048
-output_size = 80
+output_size = 150
```
Comment (blakedehaas) [inline evaluate.py:27]: Predicting 150 buckets of measurement, finer grain buckets will result in lower accuracy

File: evaluate.py [hunk 3]
```diff
@@ -111,27 +168,27 @@ def convert_to_tensor(row):
...
 plt.close()  # Close the figure to free up memory
 
-
-# Correlation plot (regular constant axes)
+# Plot absolute deviation vs ground truth
 plt.figure(figsize=(10, 8))
-h = plt.hist2d(predictions, true_values, bins=100, norm=LogNorm(), cmap='viridis', 
-               range=[[0, 15000], [0, 15000]])
+
+h = plt.hist2d(true_values, deviations, bins=100, norm=LogNorm(), cmap='viridis')
```
Comment (blakedehaas) [inline evaluate.py:174]: Should move visualizattions to visualization.py

File: train.py [hunk 4]
```diff
@@ -18,64 +18,139 @@
...
+# train_subdirs = [f.path for f in os.scandir(train_path) if f.is_dir()]
+# train_datasets = []
+# for subdir in tqdm(train_subdirs, desc="Loading train datasets"):
+#     ds = datasets.Dataset.load_from_disk(subdir)
+#     train_datasets.append(ds)
+# train_ds = datasets.concatenate_datasets(train_datasets)
+train_ds = datasets.Dataset.load_from_disk(train_path)
+val_ds = datasets.Dataset.load_from_disk("data/akebono_solar_combined_v7_chu_test")
+
+train_ds = train_ds.remove_columns(['DateTimeFormatted', 'Te1', 'Pv1', 'Te2', 'Ne2', 'Pv2', 'Te3', 'Ne3', 'Pv3', 'I1', 'I2', 'I3'])
```
Comment (blakedehaas) [inline train.py:38]: Will need to replace Te1 with Ne1 when training on electron temperature

**Review by blakedehaas — APPROVED** [review]
